File: packages/ios/Sources/Components/Card/Card.figma.swift
import SwiftUI
import Figma

struct KozmosCardBasicConnect: FigmaConnect {
    let component = KozmosCard<AnyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=87-2536"
    var variant = ["Content": "Basic"]

    @FigmaString("Body Text")
    var bodyText: String = "Use card body content for short supporting detail."

    var body: some View {
        KozmosCard {
            AnyView(
                KozmosCardContent {
                    Text(self.bodyText)
                }
            )
        }
    }
}

struct KozmosCardHeaderConnect: FigmaConnect {
    let component = KozmosCard<AnyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=87-2536"
    var variant = ["Content": "Header"]

    @FigmaString("Title Text")
    var title: String = "Card title"

    @FigmaString("Description Text")
    var description: String = "Supporting description."

    @FigmaString("Body Text")
    var bodyText: String = "Use card body content for short supporting detail."

    var body: some View {
        KozmosCard {
            AnyView(
                VStack(alignment: .leading, spacing: 0) {
                    KozmosCardHeader {
                        KozmosCardTitle(self.title)
                        KozmosCardDescription(self.description)
                    }
                    KozmosCardContent {
                        Text(self.bodyText)
                    }
                }
            )
        }
    }
}

struct KozmosCardFullConnect: FigmaConnect {
    let component = KozmosCard<AnyView>.self
    let figmaNodeUrl = "https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=87-2536"
    var variant = ["Content": "Full"]

    @FigmaString("Title Text")
    var title: String = "Card title"

    @FigmaString("Description Text")
    var description: String = "Supporting description."

    @FigmaString("Body Text")
    var bodyText: String = "Use card body content for short supporting detail."

    var body: some View {
        KozmosCard {
            AnyView(
                VStack(alignment: .leading, spacing: 0) {
                    KozmosCardHeader {
                        KozmosCardTitle(self.title)
                        KozmosCardDescription(self.description)
                    }
                    KozmosCardContent {
                        Text(self.bodyText)
                    }
                    KozmosCardFooter {
                        KozmosButton("Cancel", variant: .outline, action: {})
                        KozmosButton("Save", action: {})
                    }
                }
            )
        }
    }
}

File: packages/android/src/main/java/com/kozmos/components/Card/Card.figma.kt
package com.kozmos.components.card

import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import com.figma.code.connect.FigmaConnect
import com.figma.code.connect.FigmaProperty
import com.figma.code.connect.FigmaType
import com.figma.code.connect.FigmaVariant

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=87-2536")
@FigmaVariant("Content", "Basic")
class KozmosCardBasicConnect {
    @FigmaProperty(FigmaType.Text, "Body Text")
    val bodyText: String = "Use card body content for short supporting detail."

    @Composable
    fun ComponentExample() {
        KozmosCard {
            KozmosCardContent {
                Text(bodyText)
            }
        }
    }
}

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=87-2536")
@FigmaVariant("Content", "Header")
class KozmosCardHeaderConnect {
    @FigmaProperty(FigmaType.Text, "Title Text")
    val title: String = "Card title"

    @FigmaProperty(FigmaType.Text, "Description Text")
    val description: String = "Supporting description."

    @FigmaProperty(FigmaType.Text, "Body Text")
    val bodyText: String = "Use card body content for short supporting detail."

    @Composable
    fun ComponentExample() {
        KozmosCard {
            KozmosCardHeader {
                KozmosCardTitle(title)
                KozmosCardDescription(description)
            }
            KozmosCardContent {
                Text(bodyText)
            }
        }
    }
}

@FigmaConnect("https://figma.com/design/Yj4O8p6Y9h2Sa9zJVoAiVY?node-id=87-2536")
@FigmaVariant("Content", "Full")
class KozmosCardFullConnect {
    @FigmaProperty(FigmaType.Text, "Title Text")
    val title: String = "Card title"

    @FigmaProperty(FigmaType.Text, "Description Text")
    val description: String = "Supporting description."

    @FigmaProperty(FigmaType.Text, "Body Text")
    val bodyText: String = "Use card body content for short supporting detail."

    @Composable
    fun ComponentExample() {
        KozmosCard {
            KozmosCardHeader {
                KozmosCardTitle(title)
                KozmosCardDescription(description)
            }
            KozmosCardContent {
                Text(bodyText)
            }
            KozmosCardFooter {
                Text("Cancel")
                Text("Save")
            }
        }
    }
}
